Source/Logger.swift

```diff
@@ -32,13 +32,35 @@ import Foundation
 /// create an additional logger with a custom configuration to fit your needs.
 open class Logger {
 
-    // MARK: - Properties
+    // MARK: - Helper Types
+
+    /// Defines the two types of execution methods used when logging a message.
+    ///
+    /// Logging operations can be expensive operations when there are hundreds of messages being generated or when
+    /// it is computationally expensive to compute the message to log. Ideally, one would use the synchronous method
+    /// in development, and the asynchronous method in production. This allows for easier debugging in the development
+    /// environment, and better performance in production.
+    ///
+    /// - synchronous:  Logs messages synchronously once the recursive lock is available in serial order.
+    /// - asynchronous: Logs messages asynchronously on the dispatch queue in a serial order.
+    public enum ExecutionMethod {
+        case synchronous(lock: NSRecursiveLock)
+        case asynchronous(queue: DispatchQueue)
+    }
+
+   // MARK: - Properties
 
     /// Controls whether to allow log messages to be sent to the writers.
     open var enabled = true
 
-    /// The configuration to use when determining how to log messages.
-    open let configuration: LoggerConfiguration
+    // Log levels this logger is configured for.
+    open let logLevels: LogLevel
```

Comment:
Chatted on Slack about this and makes sense to keep as `logLevels` on in the `Logger`.